drivers/lsm6dsox/lsm6dsox.py

```diff
@@ -30,16 +30,24 @@
 import time
 from lsm6dsox import LSM6DSOX
 
-from machine import Pin, I2C
+from machine import Pin, SPI, I2C
+# Init in I2C mode.
 lsm = LSM6DSOX(I2C(0, scl=Pin(13), sda=Pin(12)))
 
+# Or init in SPI mode.
+#lsm = LSM6DSOX(SPI(5), cs_pin=Pin(10))
+
 while (True):
     print('Accelerometer: x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*lsm.read_accel()))
     print('Gyroscope:     x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*lsm.read_gyro()))
     print("")
     time.sleep_ms(100)
 """
+
 import array
+from machine import I2C
+from machine import SPI
+from machine import Pin
```

Comment:
I think it's best if this driver (or any driver for that matter) does not import hardware classes like this.  It makes the driver much more general if it doesn't rely on specific hardware in any way.

See further comments below how to do it.